templates/facetDemoScript.template.hbs

```diff
@@ -0,0 +1,165 @@
+import {
+  createPublicClient,
+  createWalletClient,
+  getContract,
+  http,
+  parseUnits,
+  Narrow,
+  zeroAddress
+} from 'viem'
+import { privateKeyToAccount } from 'viem/accounts'
+import { randomBytes } from 'crypto'
+import dotenv from 'dotenv'
+import deployments from '../../deployments/mainnet.staging.json'
+import erc20Artifact from '../../out/ERC20/ERC20.sol/ERC20.json'
+import {{camelCase name}}FacetArtifact from '../../out/{{titleCase name}}Facet.sol/{{titleCase name}}Facet.json'
+import config from '../../config/{{camelCase name}}.json'
+import { {{titleCase name}}Facet, ILiFi } from '../../typechain'
+import {
+  getEnvVar,
+  normalizePrivateKey
+} from './utils/demoScriptHelpers'
+import { getRpcUrl, getViemChain, SupportedChain } from './utils/demoScriptChainConfig'
+
+// #region ABIs
+
+const ERC20_ABI = erc20Artifact.abi as Narrow<typeof erc20Artifact.abi>
+const {{constantCase name}}_FACET_ABI = {{camelCase name}}FacetArtifact.abi as Narrow<typeof {{camelCase name}}FacetArtifact.abi>
+const EXAMPLE_ABI = [{...}] // optional define example ABI
+
+// #endregion
+
+dotenv.config()
+
+async function main() {
+  // === Set up parameters, retrieve env vars and set up clients ===
+  // `srcChain` is the source chain for the operation (change as needed).
+  // RPC URL and private key are dynamically fetched based on the chain.
+  const srcChain: SupportedChain = "mainnet"; // Set source chain
+  const RPC_URL = getRpcUrl(srcChain)
+  const PRIVATE_KEY = getEnvVar('PRIVATE_KEY')
+  const typedPrivateKey = normalizePrivateKey(PRIVATE_KEY)
```

Comment:
wondering if we could even simplify this more. 
Thinking along the lines of "I only decide which is the source chain and a helper function returns everything I need (chain, rpc, normalized private key, all viem clients, LiFiDiamond contract object, etc) as we seem to have a lot of boilerplate code in every demo script